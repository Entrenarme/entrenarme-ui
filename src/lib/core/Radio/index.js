// @flow
import * as React from 'react';
import MRadio from '@material-ui/core/Radio';
import FormControlLabel from '@material-ui/core/FormControlLabel';
import { createMuiTheme, MuiThemeProvider } from '@material-ui/core/styles';
import styled from 'styled-components';

import colors from '../../helpers/colors';
import P from '../Text/Paragraph';

const theme = createMuiTheme({
  palette: {
    primary: { main: colors.primary.default }, // Purple and green play nicely together.
    secondary: { main: colors.secondary.default }, // This is just green.A700 as hex.
  },
});

type Props = {
  label?: ?string,
};

const SRadio = styled(MRadio)`
  height: 16px !important;
  width: 16px !important;
  margin-right: 10px !important;
  svg {
    height: 16px;
  }
`;

const Radio = ({ label, ...rest }: Props) => (
  <MuiThemeProvider theme={theme}>
    <FormControlLabel
      control={<SRadio {...rest} />}
      label={label ? <P noMargin>{label}</P> : null}
    />
  </MuiThemeProvider>
);

Radio.defaultProps = {
  label: null,
};

export default Radio;
