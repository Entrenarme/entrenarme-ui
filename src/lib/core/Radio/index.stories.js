// @flow
import * as React from 'react';
import { storiesOf } from '@storybook/react';
import { action } from '@storybook/addon-actions';

import Radio from './index';

const actions = {
  onClick: action('onClick'),
};

storiesOf('Radio', module)
  .add('default', () => <Radio {...actions} />)
  .add('with label', () => <Radio {...actions} label="Label" />);
