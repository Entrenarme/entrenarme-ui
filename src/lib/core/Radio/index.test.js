import * as React from 'react';
import { render } from 'react-testing-library';

import Radio from './index';

test('renders default', () => {
  const { container } = render(<Radio />);
  expect(container.firstChild).toMatchSnapshot();
});

test('renders with label', () => {
  const { container } = render(<Radio label="Label" />);
  expect(container.firstChild).toMatchSnapshot();
});
