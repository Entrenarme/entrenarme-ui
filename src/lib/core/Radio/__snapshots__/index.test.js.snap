// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders default 1`] = `
<label
  class="MuiFormControlLabel-root-1"
>
  <span
    class="MuiButtonBase-root-19 MuiIconButton-root-13 MuiSwitchBase-root-9 MuiRadio-root-4 MuiRadio-colorSecondary-8 sc-bwzfXH dZMNYA"
  >
    <span
      class="MuiIconButton-label-18"
    >
      <svg
        class="MuiSvgIcon-root-22"
        focusable="false"
        viewBox="0 0 24 24"
        aria-hidden="true"
      >
        <path
          d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z"
        />
      </svg>
      <input
        type="radio"
        class="MuiSwitchBase-input-12"
        value=""
      />
    </span>
    <span
      class="MuiTouchRipple-root-29"
    />
  </span>
  <span
    class="MuiTypography-root-36 MuiTypography-body1-45 MuiFormControlLabel-label-3"
  />
</label>
`;

exports[`renders with label 1`] = `
<label
  class="MuiFormControlLabel-root-60"
>
  <span
    class="MuiButtonBase-root-78 MuiIconButton-root-72 MuiSwitchBase-root-68 MuiRadio-root-63 MuiRadio-colorSecondary-67 sc-bwzfXH dZMNYA"
  >
    <span
      class="MuiIconButton-label-77"
    >
      <svg
        class="MuiSvgIcon-root-81"
        focusable="false"
        viewBox="0 0 24 24"
        aria-hidden="true"
      >
        <path
          d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z"
        />
      </svg>
      <input
        type="radio"
        class="MuiSwitchBase-input-71"
        value=""
      />
    </span>
    <span
      class="MuiTouchRipple-root-88"
    />
  </span>
  <span
    class="MuiTypography-root-95 MuiTypography-body1-104 MuiFormControlLabel-label-62"
  >
    <p
      class="sc-bdVaJa dnyxcA"
    >
      Label
    </p>
  </span>
</label>
`;
